### PR TITLE
If the library required is already loaded with some other version

### LINKED ISSUE #8183: Resolving conflicts with loaded libraries when opening new ones
Suppose you open a library XXX that uses a version of some library YYY (e.g. MSL 3.2.3), but another, non-backwards compatible version of YYY (e.g. MSL 4.0.0) is already loaded. In this case, OMEdit should warned that the library being loaded uses versions of already loaded libraries which are not compatible with the required ones. It should then propose three possible actions:

1. cancel the operation: library XXX is unloaded
2. unload everything and reload library XXX from the start (in most cases this will work out of the box thanks to the uses annotations)
3. upgrade library YYY to use the already opened version of XXX, using conversion scripts (only if a conversion script is available, which would not be the case here, since there is no conversion script from MSL 4.0.0 to 3.2.3

The first option is required in case one opens the wrong version by mistake, and only figures that out when OMEdit warns him about that.

The second can be handy if one has the "wrong" default loaded version of MSL, or has messed up loading the wrong dependencies of a library first.

I'm not sure if the third option is good to have. This is what Dymola does: it starts preloading MSL 4.0.0, and then if you load a library that uses 3.2.3, it immediately offers you to upgrade it. However, upgrading a library is a big deal, because it is an irreversible operation. 

I would personally prefer that the upgrade operation is performed explicitly by first loading the library and the using right-click "convert to newer versions of used libraries". BTW, in that case, once the conversion is done, the old dependencies should be unloaded, and the new ones should be loaded instead.

### PR DESCRIPTION
**Pull request by adeas31**
### Related Issues

Fixes #8183

### Purpose

Allow the user to select what to do when wrong version of the required library is loaded.

### Approach

Allow the user to cancel the operation or unload everything and reload the class again.

### Todo

 - [x] Handle loadFile
 - [x] Handle LoadModel

### CODE REVIEW

**Comment by adeas31** [comment]
@casella this updates the warning message so instead of getting,

```
TestConversion requested package Modelica of version 3.2.3. Modelica 4.0.0 is used instead which states that it is only 
compatible with a conversion script. OpenModelica currently does not support conversion scripts and will proceed with 
potential issues as a result.
```

Now the user gets,

```
TestConversion requested package Modelica of version 3.2.3. Modelica 4.0.0 is used instead which states that it is only 
compatible with a conversion script. Use convertPackageToLibrary(TestConversion , Modelica, "4.0.0") to run the conversion script or proceed with potential issues as a result.
```

This is better when using the command line interface. However, OMEdit intercepts this message and shows the user the following dialog,

image

**Comment by perost** [comment]
Typo in the dialog: Makesure => Make sure.

Also, I think the wording on the second part is a bit weird since it's not immediately clear that you're referring to the buttons. You might want to reformulate it as e.g. "Select Cancel Operation to cancel the loading of M. Select Unload all etc.".

**Comment by casella** [comment]
The notification is fine. 

I agree with @perost that the wording of the second part of the dialog could be improved. I would suggest:

> Cancel Operation does not load class M and its dependencies
> 
> Unload All & Reload M unloads all previously loaded classes and loads M starting from a clean environment

**Comment by adeas31** [comment]
This is how it looks now,

image

File: OMEdit/OMEditLIB/OMC/OMCProxy.cpp [hunk 1]
```diff
@@ -573,9 +574,14 @@ bool OMCProxy::printMessagesStringInternal()
   /* Loop in reverse order since getMessagesStringInternal returns error messages in reverse order. */
   for (int i = errorsSize; i > 0 ; i--) {
     setCurrentError(i);
-    MessageItem messageItem(MessageItem::Modelica, getErrorFileName(), getErrorReadOnly(), getErrorLineStart(), getErrorColumnStart(), getErrorLineEnd(),
-                            getErrorColumnEnd(), getErrorMessage(), getErrorKind(), getErrorLevel());
-    MessagesWidget::instance()->addGUIMessage(messageItem);
+    const int errorId = getErrorId();
+    if (errorId == 371 || errorId == 372 || errorId == 373) {
```
Comment (adeas31) [inline OMEdit/OMEditLIB/OMC/OMCProxy.cpp:578]: @perost I am using the error ids to intercept the error and show dialog to the user. Is this the correct way? I believe better than string matching.

Comment (perost) [reply OMEdit/OMEditLIB/OMC/OMCProxy.cpp:578]: That's probably the best you can do as it is now. There's nothing stopping the ids from changing, and sometimes people change old ids to e.g. close gaps in the numbering when removing deprecated flags, but string matching has the same problem.

**Comment by casella** [comment]
@adeas31, @adrpo, any clue about the error that prevents merging in these changes?

**Comment by adrpo** [comment]
> @adeas31, @adrpo, any clue about the error that prevents merging in these changes?

@adeas31 needs to update some of the messages:
https://test.openmodelica.org/jenkins/blue/organizations/jenkins/OpenModelica/detail/PR-8764/5/tests

**Comment by adeas31** [comment]
Previously it was the text missing in the dictionary. Now some tests are failing. I will update them.

**Pull request merged**